service/history/taskProcessor.go

```diff
@@ -246,9 +248,21 @@ func (t *taskProcessor) processTaskOnce(
 	default:
 	}
 
+	var scopeIdx int
+	var err error
+
+	domainID := task.task.GetDomainID()
+
+	scopeIdx, err = task.processor.process(task)
+	scope, found := t.domainMetricsScopeCache.Get(domainID, scopeIdx)
+
+	if !found {
+		scope = t.metricsClient.Scope(scopeIdx).Tagged(t.getDomainTagByID(domainID))
```

Comment:
We should also modify `getDomainTagByID()` to return an error if it can't get domain name. And don't cache the scope in that case.